### PR TITLE
Add Bicep Support and easy setup.sh script for deployment

### PR DESCRIPTION
**Pull request by jldeen**
- Adds support to deploy all resources using Azure Bicep
- Adds support to call `./setup.sh` script for quick and easy deployment / signing of container images using Gatekeeper, Ratify, and Notation

### CODE REVIEW

File: docs/nv2-bicep.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  - AZURE_CLIENT_ID
+  - AZURE_TENANT_ID
+- Local Notation CLI needs the following local environment variables (script will take care of this):
+  - NOTATION_USERNAME
+  - NOTATION_PASSWORD
+
+### Known Issues / Currently Working on:
+
+1. ~~Add installGatekeeper Function~~
+2. ~~Add createSigningCertforKv Function~~
```
Comment (dtzar) [inline docs/nv2-bicep.md:64]: Looks like you have added these functions...

File: docs/nv2-bicep.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+  ```bash
+  ./setup.sh
+  ```
+Bash with overrides
+
+  ```bash
+  ./setup.sh my-akvrg-override 1.23.3
+  ```
+
```
Comment (dtzar) [inline docs/nv2-bicep.md:42]: Now that the script executes [all of the flow](https://github.com/Azure/notation-azure-kv/pull/17/files#diff-4209d788ad32c40cbda3c66b3de47eefb929308ca703bb77a6382625986add17R345-R353) you've got nothing really to *show* or explain.  

I suggest you break out the buildImageandSign functions so you can explain/show them (versus auto execution) or at a bare minimum put some text in here to have the user see the signed image.

**Review by dtzar — CHANGES_REQUESTED** [review]

**Comment by dtzar** [comment]
@jldeen Thanks for this PR!  A couple tweaks and I'll go ahead and merge this.

**Comment by jldeen** [comment]
@dtzar Apologies for the delay, please see latest changes and let me know if there are other updates I should add. Thank you!

**Review by dtzar — APPROVED** [review]

**Comment by dtzar** [comment]
Thank you for your work on this @jldeen!

**Pull request merged**